python_scripts/generate_test_data.py

```diff
@@ -0,0 +1,458 @@
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        # Call stored procedure with parameters for each well
+        for control in controls:
+            args = [system_name, system_run_id, control_plate_barcode, control['coordinate'], control['control']]
+            cursor.callproc('createControlPlateWellsRecord', args)
+
+        # commit changes
+        db_conn.commit()
+    except Exception as e:
+        print(e)
+    else:
+        print("Created control plate %s for run id %s" % (control_plate_barcode, system_run_id))
+    finally:
+        cursor.close()
+        db_conn.close()
+
+# Update the destination plate well after picking a control
+def update_destination_plate_well_for_control(system_name, system_run_id, destination_barcode, destination_coordinate, control_barcode, control_coordinate):
+    try:
+        # open DB connection
+        db_conn = create_database_connection()
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        # Call stored procedure with parameters
+        args = [system_name, system_run_id, destination_barcode, destination_coordinate, control_barcode, control_coordinate]
+        cursor.callproc('updateDestinationPlateWellWithControl', args)
+
+        # commit changes
+        db_conn.commit()
+    except Exception as e:
+        print(e)
+    else:
+        print("Updated destination well %s with picked control from %s coord %s" % (destination_coordinate, control_barcode, control_coordinate))
+    finally:
+        cursor.close()
+        db_conn.close()
+
+# This method is mocking the Sanger barcode creation process for destination plates
+def generate_destination_plate_barcode(system_run_id) -> str:
+    barcode = "DN%s" % str(system_run_id).zfill(8)
+    return barcode
+
+# This method is mocking the Lighthouse barcode creation process for source plates
+def generate_source_plate_barcode(system_run_id, src_plate_index) -> str:
+    barcode = "DS%s%s" % (str(system_run_id).zfill(5), str(src_plate_index).zfill(4))
+    return barcode
+
+# Choose a random coordinate from a list (and removes it from list)
+def choose_random_coordinate(well_coords) -> str:
+    index = random.randint(0, (len(well_coords) - 1))
+    coord = well_coords[index]
+    well_coords.remove(coord)
+    return coord
+
+# This mocks the LIMS source lookup API call to return a list of pickable samples
+# for a source plate
+# Sample identifiers are incremented to avoid duplicates
+def create_random_pickable_samples(system_run_id, sample_uuid_index):
+    num_pickable_samples = random.randint(1, 96)
+    pickable_samples = []
+    src_well_coords = WELL_COORDS.copy()
+    for n in range(num_pickable_samples):
+        src_coord = choose_random_coordinate(src_well_coords)
+        pickable_samples.append(
+            {
+                'source_coordinate': src_coord,
+                'sample_id': "UUID-S-%s-%s" % (str(system_run_id).zfill(5), str(sample_uuid_index).zfill(8)),
```

Comment:
I would generate a new uuid in case there is a check about format somewhere. This information is already in rna_id